### ISSUE TITLE
Stack Overflow exception with Crosshair MouseMove update

### ISSUE THREAD

**Issue by UV-Arpan** [ScottPlot/ScottPlot#1249]
## Bug Report

**Issue:** Moving the mouse a lot with Crosshairs enabled causes stack overflow due to render function call. We have 4 SignalXYConst plots with linked axes and crosshairs enabled.
![stackOverflow](https://user-images.githubusercontent.com/73156778/132444593-2a38d50f-19d8-4213-96d3-ab29e5d08dac.PNG)

**Reproducing:** Code is same as example codes for linked axes and crosshairs. 

### System Details
* **ScottPlot Version:** 4.1.17.0
* **Operating System:** Windows 10
* **Application Type:** WinForms
* **.NET Version:** NET Framework 4.8

**Comment by swharden** [comment]
Hi @UV-Arpan, thanks for reporting this! However I cannot reproduce this error based on your description. 

Can you please provide a code sample that causes this error? Here's the code that works for me.

```cs
public partial class Form1 : Form
{
    private readonly Dictionary<FormsPlot, Crosshair> CrosshairsByPlot = new Dictionary<FormsPlot, Crosshair>();

    public Form1()
    {
        InitializeComponent();

        FormsPlot[] formsPlots = { formsPlot1, formsPlot2, formsPlot3, formsPlot4 };

        foreach(FormsPlot formsPlot in formsPlots)
        {
            formsPlot.Plot.AddSignal(DataGen.Sin(51));
            Crosshair cross = formsPlot.Plot.AddCrosshair(25, .5);
            formsPlot.Render();
            formsPlot.MouseMove += FormsPlot_MouseMove;
            CrosshairsByPlot.Add(formsPlot, cross);
        }
    }

    private void FormsPlot_MouseMove(object sender, MouseEventArgs e)
    {
        FormsPlot formsPlot = (FormsPlot)sender;
        Crosshair crosshair = CrosshairsByPlot[formsPlot];
        (double x, double y) = formsPlot.GetMouseCoordinates();
        crosshair.X = x;
        crosshair.Y = y;
        formsPlot.Render();
    }
}
```

![crosshair](https://user-images.githubusercontent.com/4165489/132602889-7e311469-61c0-4eb3-a782-907e1901bf0d.gif)

If you can provide a code sample that allows me to reproduce this error I'll re-open this issue, but for now I'll close it and assume something may be wrong on your end (e.g., a typo causing a circular update loop in your code). Let me know if you figure this one out! 👍

**Comment by UV-Arpan** [comment]
Hi @swharden please give me some time. I will surely try to create a small project with this issue and send to you. Can't send you the current project directly because of IP reasons.

**Comment by sjlai1993** [comment]
Hi @swharden I also have the same issue. However, I notice that this issue only happens when using winforms and not wpf. When using wpf, this issue does not happen. Furthermore, it seems that when using winforms, crosshair will have a small delay.

I have tried this using the scottplot demo. At the left hand side is the scottplot 4.1 demo for wpf and at the right hand side is the scottplot 4.1 demo for winforms.

As shown in the video, the response of cross hair is smooth in wpf while it is laggy in winforms.

Furthermore, when I started to move the mouse quickly in winform at the 10 second time mark of the video, the winform crashes after a few seconds of quick mouse movement. This does not happen in wpf demo. 

Hope this is considered as a reproducible error as I have tried this using two of my computers.

https://user-images.githubusercontent.com/2919010/151384714-6fae2823-6c68-498d-a476-7911b8bfdbc1.mp4

**Comment by sjlai1993** [comment]
Please ignore my comment on the laggyness when using crosshair in winform. I just read the changelog where the winform demo for mousetracker have added "skipIfCurrentlyRendering: true" which explains the delay in movement.

I was wondering why the crosshair lags so much now after I redownloaded the winform demo to make the video..

**Comment by swharden** [comment]
Hi @sjlai1993, thank you for the thorough description of this bug and video demonstration!

Using your information I was able to reproduce the error on my end using the demo application available on the website, however I was unable to reproduce the error running the ScottPlot demo from the latest version of the ScottPlot source code. I believe this means that whatever was causing the crash may have been fixed since the last demo was built and published to the website.

**Want to try running the ScottPlot demo from the latest version of the source code, and let me know if the problem still remains?** Either way I'll publish a new demo to the website within the next few days (when I push the next version as a package to NuGet) and I anticipate this issue will be resolved 🤞
